torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -102,138 +112,337 @@ def get_all_unique_feature_names(self, plottable_features_only: bool = True) ->
                 Default = "", results in all the features being printed
-            module_fqn_prefix_filter (str, optional): Only includes modules with this string prefix
+            module_fqn_filter (str, optional): Only includes modules that contains this string
                 Default = "", results in all the modules in the reports to be visible in the table
 
-        Expected Use:
-            >>> model_report_visualizer.generate_table_visualization(*filters)  # outputs neatly formatted table
+        Returns a dictionary with two keys:
+            (Dict[str, Tuple[List, List]]) A dict containing two keys:
+            "tensor_level_info", "channel_level_info"
+                Each key maps to a tuple with:
+                    A list of the headers of each table
+                    A list of lists containing the table information row by row
+                    The 0th index row will contain the headers of the columns
+                    The rest of the rows will contain data
+
+        Example Use:
+            >>> mod_report_visualizer.generate_filtered_tables(
+                    feature_filter = "per_channel_min",
+                    module_fqn_filter = "block1"
+                ) # generates table with per_channel_min info for all modules in block 1 of the model
         """
-        pass
-
-    def generate_plot_info(self, feature: str, module_fqn_prefix_filter: str = "") -> Tuple[Callable, List[List[Any]]]:
+        # first get the filtered data
+        filtered_data: OrderedDict[str, Any] = self._get_filtered_data(feature_filter, module_fqn_filter)
+
+        # now we split into tensor and per-channel data
+        tensor_features: Set[str] = set()
+        channel_features: Set[str] = set()
+
+        # keep track of the number of channels we have
+        num_channels: int = 0
+
+        for module_fqn in filtered_data:
+            for feature_name in filtered_data[module_fqn]:
+                # get the data for that specific feature
+                feature_data = filtered_data[module_fqn][feature_name]
+
+                # if it is only a single value, is tensor, otherwise per channel
+                if isinstance(feature_data, torch.Tensor):
+                    # see what the size of shape is
+                    if len(feature_data.shape) == 0:
+                        # single value
+                        # means is per-tensor
+                        tensor_features.add(feature_name)
+                    else:
+                        # works means per channel
+                        channel_features.add(feature_name)
+                        num_channels = len(feature_data)
+                elif isinstance(feature_data, list):
+                    # works means per channel
+                    channel_features.add(feature_name)
+                    num_channels = len(feature_data)
+                else:
+                    # means is per-tensor
+                    tensor_features.add(feature_name)
+
```

Comment:
can we merge the per tensor cases and per channel cases into the same branch so it's clearer?

Reply:
merged